content/en/blog/_posts/2025-12-17-kubernetes-v1-35-release/index.md

```diff
@@ -0,0 +1,450 @@
+
+This work was done as part of [KEP #5538](https://kep.k8s.io/5538) led by SIG Storage.
+
+### Gang scheduling support in Kubernetes
+
+Scheduling interdependent workloads, such as AI/ML training jobs or HPC simulations, has traditionally been challenging because the default Kubernetes scheduler places Pods individually. This often leads to partial scheduling where some Pods start while others wait indefinitely for resources resulting in deadlocks and wasted cluster capacity.
+
+Kubernetes v1.35 introduces native Gang Scheduling support via the new Workload API and PodGroup concept. This feature implements an "all-or-nothing" scheduling strategy, ensuring that a defined group of Pods is scheduled only if the cluster has sufficient resources to accommodate the entire group simultaneously.
+
+The primary benefit is improved reliability and efficiency for batch and parallel workloads. By preventing partial deployments, it eliminates resource deadlocks and ensures that expensive cluster capacity is utilized only when a complete job can run, significantly optimizing the orchestration of large-scale data processing tasks.
+
+This work was done as part of [KEP #4671](https://kep.k8s.io/4671) led by SIG Scheduling.
+
+### Extended toleration operators for threshold-based placement
+
+Managing workload prioritization in Kubernetes has typically depended on PriorityClass, which orders Pods but does not account for waiting time. This static approach can lead to resource starvation for lower-priority workloads where they are perpetually preempted by higher-priority tasks making it difficult to guarantee strict Service Level Agreements (SLAs).
+
+Kubernetes v1.35 introduces SLA-aware scheduling capabilities within the Scheduling Framework. This enhancement updates the `QueueSort` plugin to optionally consider the waiting time of Pods, effectively elevating the priority of long-pending workloads to ensure they eventually get scheduled.
+
+The primary benefit is a fairer and more deterministic scheduling lifecycle. By mitigating resource starvation, operators can enforce stricter time-based SLAs and ensure that lower-priority batch jobs or background tasks make steady progress even in highly utilized clusters.
+
+This work was done as part of [KEP #5471](https://kep.k8s.io/5471) led by SIG Scheduling.
+
+### Consider terminating Pods in Deployments
+
+Updating or scaling Deployments has historically relied on an eager strategy that creates replacement Pods immediately, often ignoring those that are still terminating. This behavior can lead to resource contention and scheduling deadlocks, particularly in clusters with limited capacity, as terminating Pods continue to occupy quota while new replicas compete for the same resources.
```

Comment:
This entire paragraph is not correct. As part of the overall work towards exposing that functionality you're writing about we ONLY exposed the information about terminating replicas in the deployment. That functionality is being promoted to beta, the rest of the work under [KEP-3973](https://kep.k8s.io/3973) umbrella is still WIP.

Reply:
The title of the section is also confusing. I parsed it as "You should consider terminating the Pods in your Deployments".

Reply:
Scratch that, we see @dipesh-rawat's comment below
> I remember that this https://github.com/kubernetes/enhancements/issues/3973 could only merge some of its changes before the code freeze. Only DeploymentReplicaSetTerminatingReplicas feature gate is moved to beta for this release, although the main feature DeploymentPodReplacementPolicy is still in alpha.

> There is some additional context in latest comment post code freeze https://github.com/kubernetes/enhancements/issues/3973#issuecomment-3543463447.

> @atiratree Please feel free to jump in and help us move this forward.